src/protocol/thread/thread.ts

```diff
@@ -245,13 +247,23 @@ class _ThreadFront {
     // This is a placeholder which logs loading changes to the console.
     const sessionId = await this.waitForSession();
 
+    assert(!this.loadChangeListeners);
+    this.loadChangeListeners = [listenerCallback];
+
     client.Session.addLoadedRegionsListener((parameters: loadedRegions) => {
       console.log("LoadedRegions", parameters);
-      listenerCallback(parameters);
+      assert(this.loadChangeListeners);
+      for (const cb of this.loadChangeListeners) {
+        cb(parameters);
+      }
     });
 
     await client.Session.listenForLoadChanges({}, sessionId);
   }
+  addLoadChangeListener(cb: (params: loadedRegions) => void) {
```

Comment:
I don't like having 2 methods for adding a listener for loaded regions

Reply:
Should I just modify `listenForLoadChanges` to allow for more than one listener?

Reply:
> make listenForLoadChanges be the method that is allowed to be called more than once per program, and eliminate addLoadChangeListener

Either that, or we can change the backend (as described below) so that we can simply do `await Test.loadRegion(...)` and know that the loading has finished. In that case, we could remove this altogether (and also some code in the test harness that tries to figure out when loading a region has finished).

Reply:
That would require doing the protocol and backend changes to detect end-of-loading as we discussed below.  The alternative is somehow hooking into the model that's being updated, and get a callback whenever `app.loadedRegions` is updated.

It seems like the capability to do that would be required in general for efficient diffing of models to figure out which re-renders to issue for the UI.  Can we attach callbacks at the model level rather than listening at the protocol socket level?  If so how do I do that?

Reply:
After discussion with @hbenl we figured we could remove the `listenForLoadChanges` hooks entirely and rely purely on watching the UI to run through the test.  Tried that and it worked.